subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -1053,26 +1079,23 @@ uint32_t ull_adv_sync_time_get(const struct ll_adv_sync_set *sync,
 	return time_us;
 }
 
-uint32_t ull_adv_sync_start(struct ll_adv_set *adv,
-			    struct ll_adv_sync_set *sync,
-			    uint32_t ticks_anchor)
+uint32_t ull_adv_sync_evt_init(struct ll_adv_set *adv,
+			       struct ll_adv_sync_set *sync,
+			       struct pdu_adv *pdu)
 {
-	struct lll_adv_sync *lll_sync;
 	uint32_t ticks_slot_overhead;
 	uint32_t ticks_slot_offset;
-	uint32_t volatile ret_cb;
-	struct pdu_adv *ter_pdu;
-	uint32_t interval_us;
-	uint8_t sync_handle;
 	uint32_t time_us;
-	uint32_t ret;
 
 	ull_hdr_init(&sync->ull);
 
-	lll_sync = &sync->lll;
-	ter_pdu = lll_adv_sync_data_peek(lll_sync, NULL);
+	if (!pdu) {
+		const struct lll_adv_sync *lll_sync = &sync->lll;
 
-	time_us = ull_adv_sync_pdu_time_get(sync, ter_pdu);
+		pdu = lll_adv_sync_data_peek(lll_sync, NULL);
```

Comment:
This could probably just be

```suggestion
		pdu = lll_adv_sync_data_peek(&sync->lll, NULL);
```

without loosing readability